internal/declarative/v2/reconciler.go

```diff
@@ -463,8 +468,12 @@ func updateSyncedOCIRefAnnotation(obj Object, ref string) {
 }
 
 func pruneResource(diff []*resource.Info, resourceType string, resourceName string) []*resource.Info {
+	//nolint:varnamelen
 	for i, info := range diff {
-		obj := info.Object.(client.Object)
+		obj, ok := info.Object.(client.Object)
+		if !ok {
```

Comment:
Is hiding the conversion problem here OK? Perhaps an error should be returned instead?